### PR TITLE
fix: Layout issues during Login for iPhone and iPad when we show the keyboard - ACC - 347

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we fix the layout issues on smaller iPhones and iPads in Landscape mode when showing the keyboard. 
@netbe's initial work with the scroll view solved a lot of our problems and I made a few adjustments and improvements here and there.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/Helpers/UITextField+DoneBarButtonItem.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import UIKit
+
+extension UITextField {
+
+    func addDoneButtonOnKeyboard() {
+        let doneToolbar: UIToolbar = UIToolbar(frame: CGRect.init(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 50))
+        doneToolbar.barStyle = .default
+
+        let flexSpace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        let done: UIBarButtonItem = UIBarButtonItem(title: "Done", style: .done, target: self, action: #selector(self.doneButtonAction))
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Helpers/UITextField+DoneBarButtonItem.swift:28]: Should we localise the title?

File: Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift [hunk 2]
```diff
@@ -333,10 +340,12 @@ final class AuthenticationCredentialsViewController: AuthenticationStepControlle
         ])
 
         if shouldUseScrollView {
+            let topMargin: CGFloat = 86
```
Comment (netbe) [inline Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift:343]: minor - you could set this back to before as constant never changed.

**Review by netbe — APPROVED** [review]
Just a minor comment, otherwise your addition looks good.

However, I notice two things that need fixing (maybe another PR):

1. Configuration (showing the i button for config) is not shown when having a custom backend (with no proxy)
2. iPad layout extends to the border of the screen instead of 375 limit

File: Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift (hunk 2 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Authentication/Interface/ViewControllers/AuthenticationCredentialsViewController.swift]: Done!

File: Wire-iOS/Sources/Helpers/UITextField+DoneBarButtonItem.swift (hunk 1 above)
Comment (agisilaos) [reply Wire-iOS/Sources/Helpers/UITextField+DoneBarButtonItem.swift]: Done!

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**